### PR TITLE
Added an option to provide a custom endpoint to connect to a storage account

### PR DESCRIPTION
**Pull request by pranavsista**
…account

## Description

Added an option to provide a custom endpoint to connect to a storage account. This is a PR to demonstrate our changes and we do not plan on pushing it.

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Storage/Storage/Common/Cmdlet/NewAzureStorageContext.cs [hunk 1]
```diff
@@ -310,6 +324,12 @@ internal CloudStorageAccount GetAnonymousStorageAccount(string storageAccountNam
             return GetStorageAccountWithEndPoint(credential, storageAccountName, useHttps, storageEndpoint);
         }
 
+        internal CloudStorageAccount GetStorageAccountWithCustomEndpoint(bool useHttps, string customEndpoint)
+        {
+            StorageCredentials credential = new StorageCredentials();
```
Comment (pranavsista) [inline src/Storage/Storage/Common/Cmdlet/NewAzureStorageContext.cs:329]: This method was developed similarly to the method GetAnonymousStorageAccount() as it doesn't require a credential to connect to the storage account.

File: src/Storage/Storage/Common/Cmdlet/NewAzureStorageContext.cs [hunk 2]
```diff
@@ -462,6 +482,29 @@ private string GetTokenStrFromAccessToken(IAccessToken accessToken)
...
+        {
+            if (string.IsNullOrEmpty(endPoint))
+            {
+                throw new ArgumentException(String.Format(Resources.ObjectCannotBeNull, "CustomEndpoint"));
+            }
+
+            string blobEndpoint = endPoint;
+            string tableEndpoint = endPoint;
+            string queueEndpoint = endPoint;
+            string fileEndpoint = endPoint;
```
Comment (blueww) [inline src/Storage/Storage/Common/Cmdlet/NewAzureStorageContext.cs:503]: It's really odd to use same endpoint for blob/file/queue/table. Not sure which scenario will use it.

Comment (blueww) [reply src/Storage/Storage/Common/Cmdlet/NewAzureStorageContext.cs:503]: We should allow user to input different endpoint for different service.
It's really odd that B/T/Q/F all use same endpoint.

File: src/Storage/Storage/Common/Cmdlet/NewAzureStorageContext.cs (hunk 1 above)
Comment (blueww) [reply src/Storage/Storage/Common/Cmdlet/NewAzureStorageContext.cs:329]: The change only support customer endpoint with Anonymous credentail.
We should also support it for key/SAS/Oauth credentail.

**Comment by NoriZC** [comment]
Hi @blueww, as this draft PR hasn't been updated for long, could you please check and close it? Thanks!

**Comment by blueww** [comment]
The change has already been covered by https://github.com/Azure/azure-powershell/pull/17462.
So this PR is not needed anymore.

**Pull request closed without merging**